### PR TITLE
Docs: Update user-interfaces.rst [skip ci]

### PR DESCRIPTION
**Pull request by farrajota**
This PR updates `user-interfaces.rst` with some list text formatting and naming conventions corrections (small stuff)

### CODE REVIEW

File: docs/source/user-interfaces.rst [hunk 1]
```diff
@@ -311,14 +311,14 @@ The approaches above should suffice to convert any interface into any other.
...
     Often people are looking for APIs like ``dask.array.map_blocks`` or
-    ``dask.dataframe.map_partitions`` instead
+    ``dask.dataframe.map_partitions`` instead.
 2.  Calling ``compute()`` on Future objects.
-    Often people want the ``.result()`` method instead
-3.  Calling Numpy/Pandas functions on high-level Dask objects or
-    high-level Dask functions on Numpy/Pandas objects
+    Often people want the ``.result()`` method instead.
+3.  Calling NumPy/Pandas functions on high-level Dask objects or
+    high-level Dask functions on NumPy/Pandas objects
```
Comment (jakirkham) [inline docs/source/user-interfaces.rst]: Should we add a `.` at the end of this too?

**Comment by jakirkham** [comment]
Thanks @farrajota. One comment above. Otherwise LGTM.

File: docs/source/user-interfaces.rst (hunk 1 above)
Comment (mrocklin) [reply docs/source/user-interfaces.rst]: I'm not sure that this counts as a full sentence.  It may not stand on its own.

Comment (jakirkham) [reply docs/source/user-interfaces.rst]: There are a few things above that fall in the same category that either already had a `.` or had one added. Personally don't have strong feelings about this. Just something that caught my eye was all.

**Pull request merged**